internal/services/elasticsan/validate/elastic_san_name.go

```diff
@@ -0,0 +1,24 @@
+package validate
+
+import (
+	"fmt"
+	"regexp"
+)
+
+func ElasticSanName(i interface{}, k string) (warnings []string, errors []error) {
+	v, ok := i.(string)
+	if !ok {
+		errors = append(errors, fmt.Errorf("expected %q to be a string but it wasn't!", k))
+		return
+	}
+
+	if matched := regexp.MustCompile(`^[a-z0-9][a-z0-9_-]{1,22}[a-z0-9]$`).Match([]byte(v)); !matched {
+		errors = append(errors, fmt.Errorf("%q must be 3 to 24 characters long, must be lowercase, and can only contain alphanumeric characters, underscores, and hyphens", k))
```

Comment:
```suggestion
		errors = append(errors, fmt.Errorf("%q must be between 3 and 24 characters. It can contain only lowercase letters, numbers, underscores (_) and hyphens (-). It must start and end with a lowercase letter or number.", k))
```